core/Map.carp

```diff
@@ -154,6 +154,18 @@
           (set! m (put m k v))))
       m))
 
+  (doc to-array "Convert Map to Array of Pairs")
+  (defn to-array [m]
+    (let-do [ret (Array.zero)]
+      (for [i 0 @(n-buckets m)]
+        (let [bucket (Array.nth (buckets m) i)
+              len (Array.length (Bucket.entries bucket))
+              entries (Bucket.entries bucket)]
+          (for [j 0 len]
```

Comment:
You could use `Array.concat` here to avoid code duplication and maybe even make it a little more performant!

Reply:
good call, I was just blindly following the structure of `for-each`

Reply:
(Aside: due to a problem in my Haskell build pipeline I can’t test or reproduce any of your issues or PRs at the moment. I’m sorry!)

Reply:
I came up with

```
  (doc to-array "Convert Map to Array of Pairs")
  (defn to-array [m]
    (Array.concat
     &(Array.copy-map &(fn [b] @(Bucket.entries b)) (buckets m))))
```

but that copies each bucket twice instead of zero times like the for-loop version.

Reply:
Ah, that’s a bummer, I hadn’t thought of that. Sadly there is not `endo-map` version that changes the type, since that would potentially change the size of the array/its slots.

It’s definitely a shame, because this version looks much cleaner to me. Maybe @eriksvedang has some suggestions?